### PR TITLE
Do not use the req.SeriesLimit when consolidating fetch results

### PR DESCRIPTION
**Pull request by ryanhall07**
A previous attempt to lower the req.SeriesLimit to each database
instance, accidentally used that same lower limit when consolidating the
results from all database instances. The consolidation logic should use
the total SeriesLimit for the query.

**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/dbnode/client/session_test.go [hunk 1]
```diff
@@ -277,6 +280,67 @@ func TestIteratorPools(t *testing.T) {
...
+		index.Query{Query: idx.NewAllQuery()},
+		index.QueryOptions{
+			// set to 6 so we can test the instance series limit is 2 (6 /3 instances per replica * InstanceMultiple)
+			SeriesLimit:      6,
+			InstanceMultiple: 1,
+		})
+	require.NoError(t, err)
+	require.NotNil(t, iters)
+	// expect a series per shard.
+	require.Equal(t, 3, iters.Len())
```
Comment (ryanhall07) [inline src/dbnode/client/session_test.go:339]: verified this failed with the previous buggy code.

File: src/dbnode/client/fetch_state.go [hunk 2]
```diff
@@ -231,7 +235,9 @@ func (f *fetchState) asEncodingSeriesIterators(
 		return nil, FetchResponseMetadata{}, err
 	}
 
-	limit := f.fetchTaggedOp.requestSeriesLimit(maxInt)
```
Comment (ryanhall07) [inline src/dbnode/client/fetch_state.go:234]: this was the bug :( it was using the series limit on the req when accumulating results. this was ok previously when the req.SeriesLimit == query.SeriesLimit

File: src/dbnode/client/session.go [hunk 3]
```diff
@@ -1547,20 +1547,20 @@ func (s *session) fetchTaggedAttempt(
...
 	if err != nil {
 		s.state.RUnlock()
 		nsClone.Finalize()
 		return nil, FetchResponseMetadata{}, xerrors.NewNonRetryableError(err)
 	}
+	if opts.InstanceMultiple > 0 {
+		topo := s.state.topoMap
+		iPerReplica := int64(len(topo.Hosts()) / topo.Replicas())
+		iSeriesLimit := int64(float32(opts.SeriesLimit)*opts.InstanceMultiple) / iPerReplica
+		if req.SeriesLimit != nil && iSeriesLimit < *req.SeriesLimit {
```
Comment (wesleyk) [inline src/dbnode/client/session.go:1560]: what would `req.SeriesLimit` have previously been set to? `opts.SeriesLimit`?

File: src/dbnode/client/session_test.go [hunk 4]
```diff
@@ -277,6 +280,67 @@ func TestIteratorPools(t *testing.T) {
...
+		}, nil)
+	})
+
+	require.NoError(t, sess.Open())
+	iters, meta, err := sess.fetchTaggedAttempt(context.TODO(), ident.StringID("ns"),
+		index.Query{Query: idx.NewAllQuery()},
+		index.QueryOptions{
+			// set to 6 so we can test the instance series limit is 2 (6 /3 instances per replica * InstanceMultiple)
+			SeriesLimit:      6,
+			InstanceMultiple: 1,
```
Comment (wesleyk) [inline src/dbnode/client/session_test.go:334]: worth having a case with another instance multiple?

**Review by wesleyk — APPROVED** [review]
LGTM

File: src/dbnode/client/session.go (hunk 3 above)
Comment (ryanhall07) [reply src/dbnode/client/session.go:1560]: Yes

**Pull request merged**